File: src/core/validators/DomainEntity/DomainEntityMustContainAnIdentity.js
// @flow
import type SymbolTable from '../SymbolTable';
import { errorRuleBase } from '../ValidationRuleBase';
import { includeRuleBase } from '../ValidationRuleRepository';
import { MetaEdGrammar } from '../../../grammar/gen/MetaEdGrammar';
import { valid } from '../AbstractEntity/AbstractEntityMustContainAnIdentity';

// eslint-disable-next-line no-unused-vars
function failureMessage(ruleContext: any, symbolTable: SymbolTable): string {
  return `Domain Entity ${ruleContext.entityName().ID().getText()} does not have an identity specified.`;
}

const validationRule = errorRuleBase(valid, failureMessage);
// eslint-disable-next-line import/prefer-default-export
export const includeRule = includeRuleBase(MetaEdGrammar.RULE_domainEntity, validationRule);

File: src/core/validators/DomainEntity/DomainEntityMustContainAnIdentity.cs
using System.Linq;
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.DomainEntity
{
    public class DomainEntityMustContainAnIdentity : ValidationRuleBase<MetaEdGrammar.DomainEntityContext>
    {
        public override bool IsValid(MetaEdGrammar.DomainEntityContext context)
        {
            return context.property().Any(x => x.GetProperty().propertyComponents().propertyAnnotation().identity() != null);
        }

        public override string GetFailureMessage(MetaEdGrammar.DomainEntityContext context)
        {
            return string.Format("Domain Entity {0} does not have an identity specified.", context.entityName().ID().GetText());
        }
    }
}
